### PR TITLE
adding tests for Launchpad Landing Zone.

### PR DESCRIPTION
**Pull request by polatengin**
initial tests for the base landing zones . (This PR Is focused on launchpad.)

the following tests are implemented;

- TestLaunchpadResourceGroupIsExists
- TestLaunchpadResourceGroupIsExistsViaClient
- TestLaunchpadResourceGroupHasTags
- TestLaunchpadResourceGroupHasKeyVault
- TestLaunchpadResourceGroupHasStorageAccount
- TestLaunchpadKeyVaultHasSubscriptionIdSecret
- TestLaunchpadKeyVaultHasTenantIdSecret
- TestLaunchpadKeyVaultHasTags
- TestLaunchpadStorageAccountHasTags
- TestLaunchpadStorageAccountHasTFStateContainer

To run the tests, there is a helper bash script `run_tests.sh`

Arguments of the helper is the following;

- "LANDING_ZONES_FOLDER" -z --zones
- "CONFIG_FOLDER" -c --config
- "ENVIRONMENT" -e --environment
- "CREATE_ENV" -ce --create-env
- "DEBUG_FLAG" -d --debug

If you want to both _deploy the infrastructure_ AND _run the tests_, execute the following command;

```bash
./run_tests.sh -z ~/caf/caf-terraform-landingzones -c ~/caf/caf-terraform-landingzones-starter/configuration -e demo -ce -d
```

If you just want to run the tests against an existing infrastructure, execute the following command;

```bash
./run_tests.sh -e demo -d
```

### CODE REVIEW

File: tests/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+shArgs.arg "LANDING_ZONES_FOLDER" -z --zones PARAMETER true
+shArgs.arg "CONFIG_FOLDER" -c --config PARAMETER true
+shArgs.arg "ENVIRONMENT" -e --environment PARAMETER true
+shArgs.arg "CREATE_ENV" -ce --create-env FLAG true
+shArgs.arg "DEBUG_FLAG" -d --debug FLAG true
+
+shArgs.parse $@
+
+main() {
+  export_arm_subscription_id
```
Comment (hattan) [inline tests/run_tests.sh:22]: Minor items

* We should check for the existence of the az cli . You can use this function in scripts/lib/utils.sh
https://github.com/aztfmod/symphony/blob/master/scripts/lib/utils.sh#L8

* We should also check to ensure az is logged in, if not we can prompt the user.
You can use this function for that.
https://github.com/aztfmod/symphony/blob/master/scripts/lib/utils.sh#L16

File: tests/launchpad_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	for _, landingZone := range test.LandingZones {
+		containerName := "tfstate"
+
+		exists := azure.StorageBlobContainerExists(t, containerName, landingZone.StorageAccountName, landingZone.ResourceGroupName, test.SubscriptionID)
+
+		assert.True(t, exists, "TF State Container does not exists")
```
Comment (hattan) [inline tests/launchpad_test.go:187]: We should change the error message to

TF State Container does not exist

(no s)

**Review by hattan — APPROVED** [review]
I left a few comments, they are minor , but this looks great. Thanks!

File: tests/launchpad_test.go (hunk 2 above)
Comment (polatengin) [reply tests/launchpad_test.go]: thanks 😄

File: tests/run_tests.sh (hunk 1 above)
Comment (polatengin) [reply tests/run_tests.sh:27]: thanks for the suggestion , added 2 checks at the beginning of the main method in the correct order

**Pull request merged**